diff --git a/Makefile.in b/Makefile.in
index 3ce9afd..66d9044 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -92,7 +92,8 @@ FLAGS_TO_PASS = \
 	TAR="$(TAR)" \
 	TARFLAGS="$(TARFLAGS)" \
 	TARFILEEXT="$(TARFILEEXT)" \
-	WINDRES="$(WINDRES)"
+	WINDRES="$(WINDRES)" \
+	DESTDIR="$(DESTDIR)"
 
 # end config section
 
diff --git a/lib/Makefile.in b/lib/Makefile.in
index e524ec5..c540647 100644
--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -104,7 +104,8 @@ FLAGS_TO_PASS = \
 	TAR="$(TAR)" \
 	TARFLAGS="$(TARFLAGS)" \
 	TARFILEEXT="$(TARFILEEXT)" \
-	WINDRES="$(WINDRES)"
+	WINDRES="$(WINDRES)" \
+	DESTDIR="$(DESTDIR)"
 
 # end config section
 
diff --git a/libce/Makefile.in b/libce/Makefile.in
index 5de1633..c6fa434 100644
--- a/libce/Makefile.in
+++ b/libce/Makefile.in
@@ -198,24 +198,36 @@ lib%.a: %.o
 	$(AR) rc $@ $*.o
 	$(RANLIB) $@
 
+need-DESTDIR-compatibility = prefix libdir includedir
+.PHONY: $(need-DESTDIR-compatibility) fail-DESTDIR-compatibility
+
+$(need-DESTDIR-compatibility):
+	@test -z "$(DESTDIR)" || case "$($@)" in ?:*) \
+	  $(MAKE) --no-print-directory reject="$@" fail-DESTDIR-compatibility ;; \
+	esac
+
+fail-DESTDIR-compatibility:
+	$(error DESTDIR is not supported when $(reject) contains Win32 path `$($(reject))'; \
+	try `make install $(reject)=$(shell echo '$($(reject))' | sed s,:,:$(DESTDIR),) ...' instead)
+
 .PHONY: install install-libraries install-headers
 # install headers and libraries in a target specified directory.
 install: install-libraries install-headers
 
-install-libraries: all
-	$(mkinstalldirs) $(inst_libdir)
+install-libraries: all $(need-DESTDIR-compatibility)
+	$(mkinstalldirs) $(DESTDIR)$(inst_libdir)
 	for i in $(LIBS); do \
-		$(INSTALL_DATA) $$i $(inst_libdir)/$$i ; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(inst_libdir)/$$i ; \
 	done
 
-install-headers:
-	$(mkinstalldirs) $(inst_includedir)
+install-headers: $(need-DESTDIR-compatibility)
+	$(mkinstalldirs) $(DESTDIR)$(inst_includedir)
 	for i in $(HEADERS); do \
-		$(INSTALL_DATA) $(srcdir)/../include/$$i $(inst_includedir)/$$i ; \
+		$(INSTALL_DATA) $(srcdir)/../include/$$i $(DESTDIR)$(inst_includedir)/$$i ; \
 	done
-	$(mkinstalldirs) $(inst_includedir)/GL
+	$(mkinstalldirs) $(DESTDIR)$(inst_includedir)/GL
 	for i in $(GL_HEADERS); do \
-		$(INSTALL_DATA) $(srcdir)/../include/GL/$$i $(inst_includedir)/GL/$$i ; \
+		$(INSTALL_DATA) $(srcdir)/../include/GL/$$i $(DESTDIR)$(inst_includedir)/GL/$$i ; \
 	done
 
 # uninstall headers and libraries from a target specified directory
